internal/rbd/replicationcontrollerserver.go

```diff
@@ -290,9 +290,9 @@ func createDummyImage(ctx context.Context, rbdVol *rbdVolume) error {
 		if err != nil {
 			return err
 		}
-		dummyVol := rbdVol
+		dummyVol := *rbdVol
```

Comment:
It probably helps to have a comment about this, as the implicit deep-copy is something that is really required here. Or create a `DeepCopy()` function to make it very explicit?

Reply:
I can add a comment. if I make any code changes this should go through the testing again to cover all the cases to avoid regression.

Reply:
Do we have regression tests for this at all? If the e2e does not have a test-case, you can add the `ci/skip/e2e` label?

The current change is very fragile, and breaking seems easy. If a follow-up is planned, can you share a link to the issue?

Reply:
@nixpanic we are running the e2e to make sure nothing breaks at a cephcsi level. Yes, we can add the label here. nothing is planned an as a follow-up as code fix. https://github.com/ceph/ceph-csi/issues/2675  a tracker to remove this workaround

Reply:
regression is a manual testing for now